server/resp/src/main/java/org/infinispan/server/resp/serialization/lua/LuaResponseWriter.java

```diff
@@ -0,0 +1,285 @@
+      lua.checkStack(1);
+      if (value == null) {
+         lua.pushNil();
+      } else {
+         lua.push(value.toString());
+      }
+   }
+
+   @Override
+   public void string(byte[] value) {
+      lua.checkStack(1);
+      if (value == null) {
+         lua.pushNil();
+      } else {
+         lua.push(new String(value, StandardCharsets.US_ASCII));
+      }
+   }
+
+   @Override
+   public void integers(Number value) {
+      lua.checkStack(1);
+      if (value == null) {
+         lua.pushNil();
+      } else {
+         lua.push(value);
+      }
+   }
+
+   @Override
+   public void doubles(Number value) {
+      lua.checkStack(1);
+      if (value == null) {
+         lua.pushNil();
+      } else {
+      lua.push(value);
+      }
+   }
+
+   @Override
+   public void booleans(boolean value) {
+      lua.checkStack(1);
+      lua.push(value);
+   }
+
+   @Override
+   public void arrayEmpty() {
+      lua.checkStack(1);
+      lua.push(false);
+   }
+
+   @Override
+   public <T> void array(Collection<T> collection, JavaObjectSerializer<T> serializer) {
+      if (collection == null) {
+         nulls();
+      } else {
+         lua.checkStack(2);
+         lua.newTable();
+         int i = 1;
+         for (T o : collection) {
+            lua.push(i);
+            serializer.accept(o, this);
+            lua.setTable(-3);
+            i++;
+         }
+      }
+   }
+
+   @Override
+   public void array(Collection<?> collection, Resp3Type contentType) {
+      if (collection == null) {
+         nulls();
+      } else {
+         lua.checkStack(2);
+         lua.newTable();
+         int i = 1;
+         for (Object o : collection) {
+            lua.push(i);
+            contentType.serialize(o, this);
+            lua.setTable(-3);
+            i++;
+         }
+      }
+   }
+
+   @Override
+   public void emptySet() {
+      throw new UnsupportedOperationException("emptySet"); // Only used by COMMAND which cannot be called from Lua.
+   }
+
+   @Override
+   public void set(Set<?> set, Resp3Type contentType) {
+      if (set == null) {
+         nulls();
+      } else {
+         /*
+          * A set is represented as a lua table with two elements: 'set', containing a table
+          * with the set entries as keys and a simple true boolean as value, and `len` containing the size of the set.
+          */
+         lua.checkStack(3);
+         lua.newTable();
+         lua.push("set");
+         lua.newTable();
+         for (Object o : set) {
+            contentType.serialize(o, this);
+            lua.checkStack(1);
+            lua.push(true);
+            lua.setTable(-3);
+         }
+         lua.setTable(-3);
+         lua.push("len");
+         lua.push(set.size());
+         lua.setTable(-3);
+      }
+   }
+
+   @Override
+   public <T> void set(Set<?> set, JavaObjectSerializer<T> serializer) {
+      throw new UnsupportedOperationException(CallerId.getCallerMethodName(1));
```

Comment:
Same here, I don't know if I like the idea of leaving these little time bombs that most likely will never seen unless someone in prod or a random dev/qe dog foods and finds it later.